File: web-client/chinese-chess/src/scene/play/rule/chess_host.ts
/**
 * 棋方
 */
enum ChessHost {
    /**
     * 先手
     */
    RED = 1,

    /**
     * 后手
     */
    BLACK = 2
}

export default ChessHost;

export function reverseChessHost(host: ChessHost) {
    return host == ChessHost.BLACK ? ChessHost.RED : ChessHost.BLACK;
}

File: server/src/main/java/io/github/hulang1024/chinesechess/play/rule/ChessHost.java
package io.github.hulang1024.chinesechess.play.rule;

/**
 * 棋方
 */
public enum ChessHost {
    RED(1),
    BLACK(2);
    
    private int code;

    ChessHost(int code) {
        this.code = code;
    }

    public int code() {
        return this.code;
    }

    public ChessHost reverse() {
        return this == RED ? BLACK : RED;
    }
}
